Pidaq/UI/Stability/stability_dash/callbacks/control_callbacks.py

```diff
@@ -0,0 +1,138 @@
+
+
+@app.callback(
+    Output("save-load-connection", "value"),
+    [Input("save-profile", "n_clicks")],
+    [State("profile-name", "value"),
+     State("runtime", "value"),
+     State("timestep", "value"),
+     State("displacement-step", "value"),
+     State("velocity-step", "value"),
+     State("acceleration-step", "value")]
+)
+def save_profile(save_clicks, profile_name, runtime, timestep, displacement, velocity, acceleration):
+    
+    if not save_clicks:
+        raise dash.exceptions.PreventUpdate
+
+    folder_path = os.getcwd()
+    path_to_file = os.path.join(folder_path, "profiles.json")
+
+    if not os.path.isfile(path_to_file):
+        profiles_json = {}
+        profiles_json['profiles'] = []
+        profiles_json['profiles'].append({
+        'name': profile_name,
+        'runtime': runtime,
+        'timestep': timestep,
+        'displacement-step' : displacement,
+        'velocity-step' : velocity,
+        'acceleration-step' : acceleration
+        })
+
+        with open(path_to_file, 'w') as profiles_file:
+            json.dump(profiles_json, profiles_file, indent=4)
+
+    else:
+        with open(path_to_file, 'r') as profiles_file:
+            profiles_json = json.load(profiles_file)
+            for profile in profiles_json['profiles']:
+                if profile['name'] == profile_name:
+                    profile['runtime'] = runtime
+                    profile['timestep'] = timestep
+                    profile['displacement-step'] = displacement
+                    profile['velocity-step'] = velocity
+                    profile['acceleration-step'] = acceleration
+                    with open(path_to_file, 'w') as profiles_file:
+                        json.dump(profiles_json, profiles_file, indent=4)
+                    break
+            else:
+                profiles_json['profiles'].append({
+                'name': profile_name,
+                'runtime': runtime,
+                'timestep': timestep,
+                'displacement-step' : displacement,
+                'velocity-step' : velocity,
+                'acceleration-step' : acceleration
+                }) 
+                with open(path_to_file, 'w') as profiles_file:
+                    json.dump(profiles_json, profiles_file, indent=4)
+
+    return profile_name
+
+@app.callback(
+    [Output("load-profile", "options"),
+    Output("load-profile", "value")],
+    [Input("save-load-connection", "value"),
+    Input("delete-load-connection", "value"),
+    Input("save-profile", "n_clicks")]
+)
+def update_load_profile(profile_name, empty, save_clicks):
+    folder_path = os.getcwd()
+    path_to_file = os.path.join(folder_path, "profiles.json")
+    with open(path_to_file, 'r') as profiles_file:
+        profiles_json = json.load(profiles_file)
+        profile_labels = []
+        options=[]
+        for profile in profiles_json['profiles']:
+            profile_labels.append(profile['name'])
```

Comment:
If you implement the above system I mentioned (Although it isnt really necessary  and I probably wouldnt bother - just an optimization) - you could still iterate over all the profile names using .keys() as follows:

Recall the json:
```
{ "profiles": 
    {
    "Test-1": {

         },
    "Test-2":

    }
```


```
for name in profiles_json["profiles"].keys():
```